### PR TITLE
tidb-kubernetes: add backup and restore usage docs

### PR DESCRIPTION
**Pull request by TomShawn**
### What is changed, added or deleted? (Required)

Add documents for the CRD-based backup and restoration method for TiDB Operator 1.1 or later.

### Which TiDB version(s) do your changes apply to? (Required)

- [ ] master (the latest development version, including v4.0 changes for now)
- [x] v3.1 (TiDB 3.1 versions)
- [x] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

**If you select two or more versions from above**, to trigger the bot to cherry-pick this PR to your desired release version branch(es), you **must** add corresponding labels such as **needs-cherry-pick-3.1**, **needs-cherry-pick-3.0**, and **needs-cherry-pick-2.1**.

### What is the related PR or file link(s)?

- This PR is translated from:https://github.com/pingcap/docs-cn/pull/2055  https://github.com/pingcap/docs-cn/pull/2236  https://github.com/pingcap/docs-cn/pull/2379 https://github.com/pingcap/docs-cn/pull/2433
- Other reference link(s):

### CODE REVIEW

**Comment by TomShawn** [comment]
@yikeke @onlymellb @tennix PTAL, thanks!

File: tidb-in-kubernetes/maintain/backup-and-restore/backup-gcs.md [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+
+    {{< copyable "shell-regular" >}}
+
+    ```shell
+    kubectl create secret generic backup-demo1-tidb-secret --from-literal=password=<password> --namespace=test1
+    ```
+
+### Ad-hoc backup process
+
+Create the `Backup` CR and back up data to GSC:
```
Comment (tennix) [inline tidb-in-kubernetes/maintain/backup-and-restore/backup-gcs.md]:
```suggestion
Create the `Backup` CR and back up data to GCS:
```

File: tidb-in-kubernetes/maintain/backup-and-restore/backup-gcs.md [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+
+After creating the `Backup` CR, execute the following command to check the backup status:
+
+{{< copyable "shell-regular" >}}
+
+ ```shell
+ kubectl get bk -n test1 -owide
+ ```
+
+More `Backup` CRs are as described follows:
```
Comment (tennix) [inline tidb-in-kubernetes/maintain/backup-and-restore/backup-gcs.md]:
```suggestion
More `Backup` CRs configurations are as described follows:
```

File: tidb-in-kubernetes/maintain/backup-and-restore/backup-gcs.md [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+
+* `.spec.metadata.namespace`: the namespace where the `Backup` CR is located.
+* `.spec.from.host`: the address of the TiDB cluster to be backed up.
+* `.spec.from.port`: the port of the TiDB cluster to be backed up.
+* `.spec.from.user`: the accessing user of the TiDB cluster to be backed up.
+* `.spec.from.tidbSecretName`: the secrete of the credential needed by the TiDB cluster to be backed up.
```
Comment (tennix) [inline tidb-in-kubernetes/maintain/backup-and-restore/backup-gcs.md]:
```suggestion
* `.spec.from.tidbSecretName`: the secret of the credential needed by the TiDB cluster to be backed up.
```

**Comment by TomShawn** [comment]
All changes in this PR are migrated to https://github.com/pingcap/docs-tidb-operator/pull/1, so this PR is closed.

**Pull request closed without merging**